plugins/destination/duckdb/client/types.go

```diff
@@ -7,10 +7,45 @@ import (
 	"github.com/cloudquery/plugin-sdk/v3/types"
 )
 
-func (c *Client) SchemaTypeToDuckDB(t arrow.DataType) string {
+type listLike interface {
+	arrow.DataType
+	Elem() arrow.DataType
+}
+
+func plainWrite(dt arrow.DataType) bool {
+	return arrow.TypeEqual(dt, transformTypeForWriting(dt))
+}
+
+func transformTypeForWriting(dt arrow.DataType) arrow.DataType {
+	if dt, ok := dt.(listLike); ok {
+		return arrow.ListOf(transformTypeForWriting(dt.Elem()))
+	}
+
+	switch dt := transformType(dt).(type) {
+	case *types.UUIDType, *types.JSONType:
+		return arrow.BinaryTypes.String
+	default:
+		return dt
+	}
+}
+
+func transformSchemaForWriting(sc *arrow.Schema) *arrow.Schema {
+	fields := sc.Fields()
+	for i := range fields {
+		fields[i].Type = transformTypeForWriting(fields[i].Type)
+	}
+	md := sc.Metadata()
+	return arrow.NewSchema(fields, &md)
+}
+
+func transformType(dt arrow.DataType) arrow.DataType {
+	return duckDBToArrow(arrowToDuckDB(dt))
```

Comment:
Why wrap this function call in a function? What do we get for this added abstraction layer?

Reply:
Done in dafa00461042b75497856cabe705d2bc3c9d535e